js-api-spec/node-package-importer.node.test.ts

```diff
@@ -0,0 +1,681 @@
+            'node_modules/foo/src/sass/_styles.scss': 'a {b: c}',
+            'node_modules/foo/package.json': manifestBuilder(
+              exportsBuilder(key)
+            ),
+          });
+          dir.chdir(
+            () => {
+              const result = compileString('@use "pkg:foo";', {
+                importers: [nodePackageImporter],
+              });
+              expect(result.css).toEqualIgnoringWhitespace('a {b: c;}');
+            },
+            {changeEntryPoint: 'index.js'}
+          );
+        }));
+      it(`${key} at root without non-pathed exports`, () =>
+        sandbox(dir => {
+          dir.write({
+            'node_modules/foo/src/sass/_styles.scss': 'a {b: c}',
+            'node_modules/foo/package.json': manifestBuilder({
+              exports: {
+                [key]: './src/sass/_styles.scss',
+              },
+            }),
+          });
+          dir.chdir(
+            () => {
+              const result = compileString('@use "pkg:foo";', {
+                importers: [nodePackageImporter],
+              });
+              expect(result.css).toEqualIgnoringWhitespace('a {b: c;}');
+            },
+            {changeEntryPoint: 'index.js'}
+          );
+        }));
+
+      it(`${key} with subpath`, () =>
+        sandbox(dir => {
+          dir.write({
+            'node_modules/foo/src/sass/_styles.scss': 'd {e: f}',
+            'node_modules/foo/src/sass/_variables.scss': 'a {b: c}',
+            'node_modules/foo/package.json': manifestBuilder(
+              exportsBuilder(key)
+            ),
+          });
+          dir.chdir(
+            () => {
+              const result = compileString('@use "pkg:foo/variables";', {
+                importers: [nodePackageImporter],
+              });
+              expect(result.css).toEqualIgnoringWhitespace('a {b: c;}');
+            },
+            {changeEntryPoint: 'index.js'}
+          );
+        }));
+      it(`${key} with index`, () =>
+        sandbox(dir => {
+          dir.write({
+            'node_modules/foo/src/sass/_styles.scss': 'd {e: f}',
+            'node_modules/foo/src/sass/_variables.scss': 'g {h: i}',
+            'node_modules/foo/src/sass/colors/index.scss': 'a {b: c}',
+            'node_modules/foo/package.json': manifestBuilder(
+              exportsBuilder(key)
+            ),
+          });
+          dir.chdir(
+            () => {
+              const result = compileString('@use "pkg:foo/colors";', {
+                importers: [nodePackageImporter],
+              });
+              expect(result.css).toEqualIgnoringWhitespace('a {b: c;}');
+            },
+            {changeEntryPoint: 'index.js'}
+          );
+        }));
+    });
+    it('compiles with first conditional match found', () =>
+      sandbox(dir => {
+        dir.write({
+          'node_modules/foo/src/sass/_styles.scss': 'd {e: f}',
+          'node_modules/foo/src/sass/_variables.scss':
+            'a {from: sassCondition}',
```

Comment:
I like giving this a meaningful `from:` property, but you should also give one to the other file—otherwise you don't get any useful debugging information from the file context if the test starts failing.